recipes/fruit/all/test_package/CMakeLists.txt

```diff
@@ -4,5 +4,12 @@ project(test_package CXX)
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
 conan_basic_setup()
 
+if(NOT DEFINED CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 11)
+endif()
+if(NOT DEFINED CMAKE_CXX_STANDARD_REQUIRED)
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+endif()
+
 add_executable(${PROJECT_NAME} test_package.cpp)
 target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```

Comment:
```suggestion
target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```

Reply:
Prefer cmake target instead of global defines

Reply:
I can do that. However, the result of `CHECK_CXX_SOURCE_COMPILES` may change.

Because `CHECK_CXX_SOURCE_COMPILES` calls `try_compile()` command internally: https://cmake.org/cmake/help/latest/module/CheckCXXSourceCompiles.html
`try_compile()` is affected by `CMAKE_CXX_STANDARD`, so removing `CMAKE_CXX_STANDARD` may change the result: https://cmake.org/cmake/help/latest/command/try_compile.html#other-behavior-settings

I think we should not make this change because of that possibility.

Reply:
Yes. All (or most) of test package use set_property.

Reply:
Sorry. I was mistaken for a change request to a patch.

Accept the suggestion.

Reply:
Clean up: 02fb84584816b632f3beff0178b8724c5e4cdc00